src/editors/containers/TextEditor/components/SelectImageModal/Gallery.jsx

```diff
@@ -32,6 +34,20 @@ export const Gallery = ({
       />
     );
   }
+  if (galleryIsEmpty) {
```

Comment:
why do these have to be in the hooks, when we can just conditionally render on aspects of preexisting props? Seems just as easy to test, and way less prop passing around.

Reply:
Also leads to state-> render imbalances in your current setup.

Reply:
Ah makes sense I didn't differntiate between those well but that makes sense. State-> render imbalance just refers to a situation where the hook isn't called again because the parent component isn't updated on a state change, so the galleryIsEmpty value is stale.